sdk/core/core-paging/samples-dev/getPagedAsyncIteratorSample.ts

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+/**
+ * This sample builds a simple paging operation using getPagedAsyncIterator
+ *
+ * @summary builds a simple paging operation using getPagedAsyncIterator
+ * @azsdk-weight 40
+ */
+
+import { getPagedAsyncIterator, PagedResult, PageSettings } from "@azure/core-paging";
+
+type IsAny<T> = boolean extends (T extends never ? true : false) ? true : false
+
+function isFalse(_x: false): void { }
+function isTrue(_x: true): void { }
+
+export async function main() {
+  const totalElementsCount = 100;
+  const collection = Array.from(Array(totalElementsCount), (_, i) => i + 1);
+  const pagedResult: PagedResult<number[], PageSettings, number> = {
+    firstPageLink: 0,
+    async getPage(pageLink, maxPageSize) {
+      const top = maxPageSize || 5;
+      if (pageLink < collection.length) {
+        return Promise.resolve({
+          page: collection.slice(pageLink, Math.min(pageLink + top, collection.length)),
+          nextPageLink: top < collection.length - pageLink ? pageLink + top : undefined,
+        });
+      } else {
+        throw new Error("should not get here");
+      }
+    },
+  };
+  // instantiate the generic type variables so that TElement does not default to unknown
+  const iterator = getPagedAsyncIterator<number, number[], PageSettings, number>(pagedResult);
+
+  // iterating over pages with for-await
+  for await (const page of iterator.byPage({ maxPageSize: 5 })) {
+    for (const element of page) {
+      const elementHasTypeAny: IsAny<typeof element> = false;
+      isFalse(elementHasTypeAny);
```

Comment:
I am adding these to make sure the type of `element` is not `any`. The first line is enough for this purpose, but I had to add the second to trick TS into thinking the variable `elementHasTypeAny` is "used".

Reply:
Maybe put a comment to this effect in the code?

Reply:
You could wrap this up into an empty assertNotAny function call maybe?

```
function assertNotAny<T extends (IsAny<T> extends true ? never : any)>(t: T): void { }
```

could work, assuming the intent is that the compiler yells at you if it is any for some reason. But I'm not sure this needs to be in the samples?

Reply:
yeah, couldn't it be a test?

Reply:
@bterlson nice, I used `assertNotAny`!

@xirzec Good point! I converted it to tests with descriptive names.